.github/workflows/ci.yml

```diff
@@ -56,23 +57,28 @@ jobs:
           else
             echo "No Docker credentials, skipping login"
           fi
-      - name: Build dependabot-core image
+      - name: Log in to GHCR
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+      - name: Build dependabot-core image for branch
+        env:
+          DOCKER_BUILDKIT: 1
         run: |
-          DOCKER_BUILDKIT=1 docker build \
-            -t "$CORE_IMAGE:latest" \
-            -t "$CORE_IMAGE:branch--$BRANCH_REF" \
+          docker build \
+            -t "$CORE_BRANCH_IMAGE:$BRANCH_REF" \
             --build-arg BUILDKIT_INLINE_CACHE=1 \
             --cache-from "$BASE_IMAGE" \
-            --cache-from "$CORE_IMAGE:branch--$BRANCH_REF" \
             --cache-from "$CORE_IMAGE:latest" \
+            --cache-from "$CORE_BRANCH_IMAGE:$BRANCH_REF" \
```

Comment:
This is an intentional format change since the `branch` name has folded into the container repository.

Old: `dependabot/dependabot-core:branch--refs-pull-4511-merge`
New: `ghcr.io/dependabot/dependabot-core-branch:refs-pull-4511-merge`